### PR TITLE
[SDK:RTL] RtlValidAcl(): Improve ACL error messages. CORE-11506

### PR DESCRIPTION
**Pull request by SergeGautherie**
Improve debug log a bit.

JIRA issue: [CORE-11506](https://jira.reactos.org/browse/CORE-11506)

### CODE REVIEW

File: sdk/lib/rtl/acl.c [hunk 1]
```diff
@@ -847,21 +847,21 @@ RtlValidAcl(IN PACL Acl)
         if ((Acl->AclRevision < MIN_ACL_REVISION) ||
             (Acl->AclRevision > MAX_ACL_REVISION))
         {
-            DPRINT1("Invalid ACL revision\n");
+            DPRINT1("Invalid ACL revision: %hhu\n", Acl->AclRevision);
```
Comment (ThFabba) [inline sdk/lib/rtl/acl.c:850]: `%hhu` (or even `%hu`) is not a thing, integers get promoted. Just %u will be fine.

[the remark this answers is not in the document]
Comment (ThFabba) [reply sdk/lib/rtl/acl.c:850]: The type of this is UCHAR, so `%lu` isn't right either.

[the remark this answers is not in the document]
Comment (HBelusca) [reply sdk/lib/rtl/acl.c:850]: Oh yes I misread it.

File: sdk/lib/rtl/acl.c [hunk 2]
```diff
@@ -847,21 +847,21 @@ RtlValidAcl(IN PACL Acl)
...
-            DPRINT1("Invalid ACL size\n");
+            DPRINT1("Invalid ACL size: %hu\n", Acl->AclSize);
             _SEH2_YIELD(return FALSE);
         }
 
         /* And that it's big enough */
         if (Acl->AclSize < sizeof(ACL))
         {
-            DPRINT1("Invalid ACL size\n");
+            DPRINT1("Invalid ACL size: %hu\n", Acl->AclSize);
```
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply sdk/lib/rtl/acl.c:864]: No.

File: sdk/lib/rtl/acl.c [hunk 3]
```diff
@@ -847,21 +847,21 @@ RtlValidAcl(IN PACL Acl)
...
-            DPRINT1("Invalid ACL revision\n");
+            DPRINT1("Invalid ACL revision: %hhu\n", Acl->AclRevision);
             _SEH2_YIELD(return FALSE);
         }
 
         /* Next, validate that the ACL is USHORT-aligned */
         if (ROUND_DOWN(Acl->AclSize, sizeof(USHORT)) != Acl->AclSize)
         {
-            DPRINT1("Invalid ACL size\n");
+            DPRINT1("Invalid ACL size: %hu\n", Acl->AclSize);
```
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply sdk/lib/rtl/acl.c:857]: No.

File: sdk/lib/rtl/acl.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply sdk/lib/rtl/acl.c:850]: No reason for hexa: it's a simple number.

[the remark this answers is not in the document]
Comment (HBelusca) [reply sdk/lib/rtl/acl.c:850]: 02u then.

File: sdk/lib/rtl/acl.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (HBelusca) [reply sdk/lib/rtl/acl.c:864]: That's what we use in other places of kernel for displaying an unsigned long (alternatively, `%u`).

[the remark this answers is not in the document]
Comment (SergeGautherie) [reply sdk/lib/rtl/acl.c:864]: `AclSize` is a `USHORT`...

File: sdk/lib/rtl/acl.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply sdk/lib/rtl/acl.c:850]: No reason for `02` either: it's a simple number.

[the remark this answers is not in the document]
Comment (HBelusca) [reply sdk/lib/rtl/acl.c:850]: It's a UCHAR so how can the DPRINT know it has to only read 1 byte? If you use %c you'll get it displayed as a character instead (which is not what you want).

File: sdk/lib/rtl/acl.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (HBelusca) [reply sdk/lib/rtl/acl.c:864]: oh yes, I was not reading the correct structure.

File: sdk/lib/rtl/acl.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThFabba) [reply sdk/lib/rtl/acl.c:850]: Variadic arguments get promoted to int if they are smaller than that. `%u` is the correct choice here.

[the remark this answers is not in the document]
Comment (HBelusca) [reply sdk/lib/rtl/acl.c:850]: Thanks for the explanation!

Comment (SergeGautherie) [reply sdk/lib/rtl/acl.c]: As you prefer. Done.

**Pull request merged**